src/OpenGL/CGLContext.cs

```diff
@@ -30,9 +30,17 @@
 
 using ObjCRuntime;
 using Foundation;
+using System.Runtime.Versioning;
 
 namespace OpenGL {
-	[Deprecated (PlatformName.MacOSX, 10, 14, message : "Use 'Metal' Framework instead.")]
+	#if !NET
+			[Deprecated (PlatformName.MacOSX, 10, 14, message: "Use 'Metal' Framework instead.")]
+	#else
+			[UnsupportedOSPlatform ("macos10.14")]
+	#if MONOMAC
+			[Obsolete ("Starting with macos10.14 Use 'Metal' Framework instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+	#endif
+	#endif
```

Comment:
for future PR do not indent the `#if`
that will make the `[Deprecated ...]` shows as not changed (by default) and make PR easier to review

Reply:
unless this is how @mandel-macaque macros/snippets are working ?

Reply:
That was manual, the snippets don't do that

Reply:
Sorry, don't know why I did that haha